src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -130,8 +130,16 @@
   [master-ns :- schema/Keyword
    config-route]
   (cond
-    (and (map? config-route) (contains? config-route :master-routes))
-    (:master-routes config-route)
+    ;; if the route config is a map, we need to determine whether it's the
+    ;; new-style multi-server config (where there will be a `:route` key and a
+    ;; `:server`, key), or the old style where there is a single key that is
+    ;; assumed to be our hard-coded route id (`:master-routes`).
+    ;; It should be possible to delete this hack (perhaps this entire function)
+    ;; when we remove support for legacy routes.
+    (and (map? config-route) (or (contains? config-route :route)
```

Comment:
I'm fine with waiting until the need arises.  Like you mentioned, I'm hoping that the legacy routes service would die long before / if we'd ever have a need to host multiple route-ids under the master-service.  Seems like if that were to come up that any failures by the legacy routes service to handle it would be pretty obvious.